### DISCUSSION TITLE
How can I avoid ExpectNoMsg calls without cluttering the code to achieve faster testing times

### DISCUSSION THREAD

**Discussion by Chaosflo** [akkadotnet/akka.net#7131] [Q&A]
I have the following 'issue' with test time execution and I need some ideas.
The thing is the test execution time get's risen by a exceptional amount when using ExpectNoMsg.
Which is totally expected, but I really want to shorten the time here.

I have a few ideas to omit this but I don't really know if they are any good?

1. One would be writing a log where I can be sure that nothing will happen anymore.
2. Sent another message after the initial one, which should then emit a dead letter, with this I can be sure the the first one is finished completely right?

The only thing I need to know is if the receive handler in the actor has finished doing something.

Thank you in advance for your input and ideas =) 
Maybe its just a stupid question too

**Comment by Aaronontheweb** [comment]
> The thing is the test execution time get's risen by a exceptional amount when using ExpectNoMsg.

Yep, this is to be expected (pun intended) - testing for evidence of absence with `ExpectNoMsg` is slow. Using the `EventFilter` to look for a `DeadLetter` might be a better option - the other thing you can do, if the actor shuts itself down after processing, is to use the `Watch` / `ExpectTerminated` features - that will send a signal that the actor is dead and won't process any more messages.

**Reply by Chaosflo** [reply]
Thanks for idea! 
I will try it out =)

**Reply by Aaronontheweb** [reply]
Let me know if it solves your problem
